docs/dependency-injection-liftweb-scala.adoc

```diff
@@ -0,0 +1,382 @@
+that allow basic DI without resorting to entire DI frameworks.
+
+Scala allows the use of http://jonasboner.com/real-world-scala-dependency-injection-di/[cake pattern]
+that can provide DI by composing traits.
+
+Although the cake pattern is helpful, it doesn't provide complete DI functionality.
+The cake pattern allows you to compose complex classes out of Scala traits, but the it's less helpful
+in terms of allowing you to make dynamic choices about which combination of cake to vend in a given situation.
+Lift provides extra features that complete the dependency injection puzzle.
+
+== Why Lift DI
+
+Many people come to Scala from a Java background, so it's only natural to want to use Guice/Spring,
+which do a great job in the java world. And we can https://groups.google.com/forum/#!topic/liftweb/_lleL2xpCFU[use those],
+but there's a better, more direct way that's less magical.
+
+Lift's philosophy is to keep things as simple as possible (and no simpler), and allow full control
+to the developers over how things work. To paraphrase David Pollak,
+"The abstraction turtles have to end somewhere" and  lift's DI features end those turtles quite
+early.
+
+== Lift DI and various dependency scopes (e.g. from Spring)
+
+Lift's DI allows all the
+https://docs.spring.io/spring/docs/current/spring-framework-reference/html/beans.html#beans-factory-scopes[four most commonly used scopes] that
+are possible with spring:
+https://docs.spring.io/spring/docs/current/spring-framework-reference/html/beans.html#beans-factory-scopes-singleton[singleton],
+https://docs.spring.io/spring/docs/current/spring-framework-reference/html/beans.html#beans-factory-scopes-prototype[prototype], https://docs.spring.io/spring/docs/current/spring-framework-reference/html/beans.html#beans-factory-scopes-request[request],
+and https://docs.spring.io/spring/docs/current/spring-framework-reference/html/beans.html#beans-factory-scopes-session[session]. The others can be achieved
+relatively easily if you want.
+
+== Getting started
+
+Lift's DI is based on two main traits https://liftweb.net/api/26/api/index.html#net.liftweb.util.Injector[`Injector`]
+and https://liftweb.net/api/26/api/index.html#net.liftweb.util.Maker[`Maker`]. However, most of the time, you won't need to
+deal with them directly. The elements that you would use are:
+
+. https://liftweb.net/api/26/api/index.html#net.liftweb.http.Factory[`Factory`] trait
+. https://github.com/lift/framework/blob/5033c8798d4444f81996199c10ea330770e47fbc/web/webkit/src/main/scala/net/liftweb/http/Factory.scala#L37[`FactoryMaker`], which is an abstract class inside
+ inside the [`Factory`] trait.
+. https://liftweb.net/api/26/api/index.html#net.liftweb.util.SimpleInjector$Inject[`Inject`], an abstract class within the
+ `SimpleInjector` trait.
+
+`FactoryMaker` and `Injector` serve the same purpose, with the former
+having more features. The difference is important. Notes on this are
```

Comment:
I would make this read `The main difference is in performance, and is discussed in the <<section-tag,section on differences between FactoryMaker and Inject>>` (using asciidoc [internal reference syntax](http://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#links)).